src/services/configServices/SettingsService.js

```diff
@@ -140,12 +140,15 @@ class SettingsService {
     const clonedCurrentData = _.cloneDeep(currentData)
     Object.keys(updatedData).forEach((field) => {
       if (field === "social_media") {
+        // Set social_media field if it doesn't exist
+        if (!clonedCurrentData[field]) clonedCurrentData[field] = {}
         const socials = updatedData[field]
         Object.keys(socials).forEach((social) => {
           if (!socials[social]) {
-            delete clonedCurrentData[field][social]
+            if (clonedCurrentData[field][social])
+              delete clonedCurrentData[field][social]
           } else {
-            clonedCurrentData[field] = updatedData[field]
+            clonedCurrentData[field][social] = updatedData[field][social]
```

Comment:
not sure if out of scope but this has 4 levels of `if/else` and lacks comments so it's q difficult to follow ._. perhaps we should try to make this more readable **if** we have time/it's easy

Reply:
Hmm I've added comments and removed one layer of unnecessary conditionals in https://github.com/isomerpages/isomercms-backend/pull/496/commits/2103a6a94c99c7f1586caf9cee969a85ea6cb34c - lmk if it's still unclear!

Reply:
after reading through this, i feel that we should extract this pattern out into a function, perhaps called `cloneDeepNonEmpty` (pls feel free to suggest a better naming ._.)

this is because what this is trying to do is basically trying to clone non empty fields for both the top level object as well as `socialMediaData`

Reply:
https://github.com/isomerpages/isomercms-backend/pull/496/commits/cc11c5f7d34b260acba0ddfefd7eb10c6ab4ecc2